### PR TITLE
Added tests for QAN

### PR DESCRIPTION
**Pull request by Kartikeya99**

### CODE REVIEW

File: codecept.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
     "postgresqlOverviewPage": "./tests/pages/postgresqlOverviewPage.js",
     "mongodbOverviewPage": "./tests/pages/mongodbOverviewPage.js",
     "summaryDashboardPage": "./tests/pages/summaryDashboardPage.js",
-    "trendsDashboardPage": "./tests/pages/trendsDashboardPage.js"
+    "trendsDashboardPage": "./tests/pages/trendsDashboardPage.js",
+    "qanPage" : "./tests/pages/qanPage.js"
```
Comment (puneet0191) [inline codecept.json]: QueryAnalyticsPage.js

File: tests/pages/qanPage.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const I = actor();
+
+module.exports = {
+
+    fields : {
+        table: "//app-qan-table//div[@class='ps-content']",
+        filter: "//app-qan-filter//div[@class='ps-content']",
+        search: "//app-qan-search//input",
+        pagination: "//div[@class='pagination-tools']//ul[@class='ngx-pagination']",
+        pmmImage: "//footer/img",
```
Comment (puneet0191) [inline tests/pages/qanPage.js]: why?

File: tests/pages/qanPage.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        pmmVersion: "//footer/small",
+        paginationArrow: "(//ul[@class='ngx-pagination']/li)[last()]",
+        addColumn: "//button[@class='add-column-btn']",
+        total: "//span[contains(text(), 'TOTAL')]",
+        columns: "//tbody//app-qan-table-header-cell",
+
```
Comment (puneet0191) [inline tests/pages/qanPage.js]: reason for this empty line?

File: tests/pages/qanPage.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        I.seeElement(this.fields.table);
+        I.seeElement(this.fields.filter);
+        I.seeElement(this.fields.search);
+        I.seeElement(this.fields.pagination);
+        I.seeElement(this.fields.pmmImage);
+        I.seeElement(this.fields.pmmVersion);
+        I.seeElement(this.fields.addColumn);
+        I.seeElement(this.fields.total);
+
+        I.seeElement(this.fields.columns + "[1]"); // load
```
Comment (puneet0191) [inline tests/pages/qanPage.js]: what is the need to hardcode element in the array?

File: tests/pages/qanPage.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        I.seeElement(this.fields.columns + "[3]"); // time
+    },
+
+    checkForPagination() {
+        I.click(this.fields.paginationArrow);
+        I.wait(10);
+        I.seeElement(this.fields.columns + "[1]"); // load
+    },
+
+    // checkChangeNumResults() {
```
Comment (puneet0191) [inline tests/pages/qanPage.js]: Why this is commented?

File: tests/verifyQAN_test.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+Before((I, loginPage) => {
+    I.amOnPage(loginPage.url);
+    loginPage.login("admin", "admin");
+});
+
+Scenario('Open the QAN Dashboard and Test for Elements', async (I, adminPage, qanPage) => {
+    await adminPage.navigateToDashboard("QAN", "_PMM Query Analytics");
+
+    within({frame: "//div[@class='panel-content']//iframe"}, () => {
```
Comment (puneet0191) [inline tests/verifyQAN_test.js]: Good Job!, does the frame have a name?

File: tests/pages/qanPage.js (hunk 3 above)
Comment (Kartikeya99) [reply tests/pages/qanPage.js]: It was a mental note for me to mark that the number of the results section is yet to be tested.

File: tests/pages/qanPage.js (hunk 5 above)
Comment (Kartikeya99) [reply tests/pages/qanPage.js]: I was coming across a bug I had to search more about and it isn't completely tested. Will uncomment as soon as I figure out the problem.

File: tests/pages/qanPage.js (hunk 4 above)
Comment (Kartikeya99) [reply tests/pages/qanPage.js]: There were three elements from that xpath - `//tbody//app-qan-table-header-cell` and I had to make sure each one is present.

**Pull request merged**